osu.Server.Spectator/Hubs/ScoreUploader.cs

```diff
@@ -101,14 +102,26 @@ public async Task Flush()
 
                         try
                         {
-                            if (scoreId != null)
+                            if (dbScore == null)
                             {
-                                item.Score.ScoreInfo.OnlineID = scoreId.Value;
-                                await scoreStorage.WriteAsync(item.Score);
-                                await db.MarkScoreHasReplay(item.Score);
-                            }
-                            else
                                 Console.WriteLine($"Score upload timed out for token: {item.Token}");
+                                return;
+                            }
+
+                            if (!dbScore.ScoreInfo.Passed)
+                                return;
+
+                            var updatedScore = new Score
+                            {
+                                Replay = item.Score.Replay,
+                                ScoreInfo = dbScore.ScoreInfo.ToScoreInfo(item.Score.ScoreInfo.Mods, item.Score.ScoreInfo.BeatmapInfo)
```

Comment:
Is there a reason why this is preferring the db-retrieved `ScoreInfo` over the already-existing local one? Seems generally arbitrary.

Reply:
The alternative is to manually copy over the `Passed` value from the databased model. Does that sit better with you?

For me, if we have to do that for `Passed` then it seems as something we would have to do over again in the future for other properties, and I feel uneasy about that.